src/many/src/server/module/_9_account.rs

```diff
@@ -0,0 +1,443 @@
+
+    #[n(2)]
+    pub features: features::FeatureSet,
+}
+
+#[many_module(name = AccountModule, id = 9, namespace = account, many_crate = crate)]
+pub trait AccountModuleBackend: Send {
+    /// Create an account.
+    fn create(&mut self, sender: &Identity, args: CreateArgs) -> Result<CreateReturn, ManyError>;
+
+    /// Set the name/description of an account.
+    fn set_description(
+        &mut self,
+        sender: &Identity,
+        args: SetDescriptionArgs,
+    ) -> Result<EmptyReturn, ManyError>;
+
+    /// List all the roles supported by an account.
+    fn list_roles(
+        &self,
+        sender: &Identity,
+        args: ListRolesArgs,
+    ) -> Result<ListRolesReturn, ManyError>;
+
+    /// Get roles associated with an identity for an account.
+    fn get_roles(&self, sender: &Identity, args: GetRolesArgs)
+        -> Result<GetRolesReturn, ManyError>;
+
+    /// Add roles to identities for an account.
+    fn add_roles(&self, sender: &Identity, args: AddRolesArgs) -> Result<EmptyReturn, ManyError>;
+
+    /// Remove roles from an identity for an account.
+    fn remove_roles(
+        &self,
+        sender: &Identity,
+        args: RemoveRolesArgs,
+    ) -> Result<EmptyReturn, ManyError>;
+
+    /// Returns the information related to an account.
+    fn info(&self, sender: &Identity, args: InfoArgs) -> Result<InfoReturn, ManyError>;
+
+    /// Delete an account.
+    fn delete(&self, sender: &Identity, args: DeleteArgs) -> Result<EmptyReturn, ManyError>;
+
+    /// Add additional features to an account.
+    fn add_features(
+        &self,
+        sender: &Identity,
+        args: AddFeaturesArgs,
+    ) -> Result<EmptyReturn, ManyError>;
+}
+
+#[cfg(test)]
+mod module_tests {
+    use super::*;
+    use crate::server::module::testutils::call_module;
+    use std::sync::{Arc, Mutex};
+
+    struct AccountImpl(pub AccountMap);
+    impl std::default::Default for AccountImpl {
+        fn default() -> Self {
+            Self(AccountMap::new(
+                Identity::from_bytes(
+                    &hex::decode("0102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D")
+                        .unwrap(),
+                )
+                .unwrap(),
+            ))
+        }
+    }
+
+    impl super::AccountModuleBackend for AccountImpl {
+        fn create(
+            &mut self,
+            _sender: &Identity,
+            args: CreateArgs,
+        ) -> Result<CreateReturn, ManyError> {
+            let (id, _) = self.0.insert(Account::create(args))?;
+            Ok(CreateReturn { id })
+        }
+
+        fn set_description(
+            &mut self,
+            _sender: &Identity,
+            args: SetDescriptionArgs,
+        ) -> Result<EmptyReturn, ManyError> {
+            let mut account = self
+                .0
+                .get_mut(&args.id)
+                .ok_or_else(|| errors::unknown_account(args.id.to_string()))?;
+
+            account.name = Some(args.description);
```

Comment:
We should probably stick to a single nomenclature, either `name` or `description`.